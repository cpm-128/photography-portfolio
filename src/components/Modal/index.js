// child of PhotoList so that the modal can open
// modal must close at the child (here)

import React from 'react';

// currentPhoto is defined in PhotoList
// pass the onClose as a prop from the PhotoList component
const Modal = ({ currentPhoto, onClose }) => {

    // deconstruct the prop that is currentPhoto
    const {name, category, description, index } = currentPhoto;

    return (
        // basic modal with a backdrop and container
        <div className="modalBackdrop">
            <div className="modalContainer">
                <h3 className="modalTitle">{name}</h3>
                <img src={require(`../../assets/large/${category}/${index}.jpg`)} alt="current category" />
                <p>{description}</p>
                <button type="button" onClick={onClose}>
                    Close this modal
                </button>
            </div>
        </div>
    )
};

export default Modal;