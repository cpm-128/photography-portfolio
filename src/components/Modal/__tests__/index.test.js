import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

// declare the props
const mockToggleModal = jest.fn();

const currentPhoto = {
    name: 'Pug smile',
    category: 'portraits',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    index: 3
};

afterEach(cleanup);

describe('Modal is rendering', () => {

    // 1st test: baseline to verify the component is rendering
    test('renders', () => {
        render(<Modal
            onClose={mockToggleModal}
            currentPhoto={currentPhoto}
        />);
    });

    // 2nd test: compare snapshot to DOM node structure
    test('matches snapshot DOM node structure', () => {
        const { asFragment } = render(<Modal
            onClose={mockToggleModal}
            currentPhoto={currentPhoto}
        />);
        expect(asFragment()).toMatchSnapshot();
    })

})

describe('click event', () => {
    test('calls onClose handler', () => {
        // Arrange: Render Modal
            const { getByText } = render(<Modal
                onClose={mockToggleModal}
                currentPhoto={currentPhoto}
            />);
        // Act: Simulate click event
            fireEvent.click(getByText('Close this modal'));
        // Assert: expected matcher
            expect(mockToggleModal).toHaveBeenCalledTimes(1);
    });
})