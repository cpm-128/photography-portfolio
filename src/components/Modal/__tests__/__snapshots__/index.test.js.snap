// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal is rendering matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Pug smile
      </h3>
      <img
        alt="current category"
        src="3.jpg"
      />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
